rn/Teacher/src/modules/assignments/__tests__/AssignmentList.test.js

```diff
@@ -61,409 +60,323 @@ beforeEach(() => {
+    let groupOne = templates.assignmentGroup({
+      id: '1',
+      assignments: [ templates.assignment({ id: '1', name: 'One' }) ],
+    })
+    let groupTwo = templates.assignmentGroup({
+      id: '2',
+      assignments: [ templates.assignment({ id: '2', name: 'Two' }) ],
+    })
+    let gradingPeriod = templates.gradingPeriod({
+      assignmentRefs: ['1'],
+    })
+    let tree = shallow(
+      <AssignmentList
+        {...defaultProps}
+        assignmentGroups={[groupOne, groupTwo]}
+        gradingPeriods={[gradingPeriod]}
+      />
+    )
+    ActionSheetIOS.showActionSheetWithOptions = jest.fn((options, fn) => fn(0))
+    let filter = tree.find('[testID="assignment-list.filter"]')
+    filter.simulate('press')
+
+    let list = tree.find('SectionList')
+    let sections = list.props().sections
+    let row = shallow(list.props().renderItem({ item: sections[0].data[0], index: 0 })).find('Row')
+
+    expect(tree.find('Heading1').props().children).toEqual(gradingPeriod.title)
+    expect(sections.length).toEqual(1)
+    expect(row.prop('title')).toEqual('One')
+
+    ActionSheetIOS.showActionSheetWithOptions = jest.fn((options, fn) => fn(1))
+    filter.simulate('press')
+    expect(tree.find('SectionList').props().sections.length).toEqual(2)
+  })
 
-  expect(tree.find('ActivityIndicator').length).toEqual(1)
-})
+  it('will call refreshlist with the grading period id when it has no assignmentRefs', () => {
+    let tree = shallow(
+      <AssignmentList
+        {...defaultProps}
+      />
+    )
+    tree.instance().updateFilter(0)
+    expect(defaultProps.refreshAssignmentList).toHaveBeenCalledWith(course.id, gradingPeriod.id, true)
+  })
 
-test('calls getGradesForGradingPeriod when a filter is updated', async () => {
-  let grades = templates.enrollment().grades
-  let promise = Promise.resolve(grades)
-  getGradesForGradingPeriod.mockReturnValueOnce(promise)
+  it('removes filter when clear button pressed', () => {
+    let tree = shallow(
+      <AssignmentList {...defaultProps} />
+    )
+
+    ActionSheetIOS.showActionSheetWithOptions = jest.fn((options, fn) => fn(0))
+    let filter = tree.find('[testID="assignment-list.filter"]')
+    filter.simulate('press')
+    let clearButton = tree.find('[testID="assignment-list.filter"]')
+    expect(clearButton.props().children).toEqual('Clear filter')
+    clearButton.simulate('press')
+    tree.update()
+    expect(tree.find('Heading1').props().children).toEqual('All Grading Periods')
+  })
 
-  let tree = shallow(
-    <AssignmentList {...defaultProps} showTotalScore />
-  )
+  it('selects first item on regular horizontal trait collection', () => {
+    let traits = { window: { horizontal: 'regular' } }
+    defaultProps.navigator.traitCollection = jest.fn(fn => fn(traits))
+    let screen = shallow(
+      <AssignmentList {...defaultProps} />
+    )
+    screen.props().onTraitCollectionChange()
+    expect(defaultProps.navigator.show).toHaveBeenCalled()
```

Comment:
Since we are testing that the first item is shown, we should assert that the correct screen was shown.

Reply:
Ah yes, good catch.